### PR TITLE
Refactor documentation

### PR DESCRIPTION
**Pull request by nafur**
This PR moves some stuff around in our documentation. Most notably, it moves some sections out of the "Binary documentation" to become their own top-level sections. While doing so, it refactors a few things in options and statistics to be more agnostic to the way cvc5 is used. To allow for `command-output` being used in regular (not auto-generated) documentation, we add a new extension that takes care of injecting the build folder into a new wrapper `run-command`.

### CODE REVIEW

**Comment by nafur** [comment]
I think this is mostly what we have discussed offline. Please have a look and let me know what should still be changed. The updated documentation should be available at https://cvc5.github.io/docs-ci/docs-pr8288/ shortly after the CI is done with this PR...

File: docs/binary/binary.rst [hunk 1]
```diff
@@ -3,8 +3,12 @@ Binary Documentation
 
 The easiest way to use cvc5 is usually to invoke the cvc5 binary via the command line.
 The :doc:`quickstart guide <quickstart>` gives a short introduction on how to use cvc5 via the SMT-LIBv2
-interface, but cvc5 also supports other :doc:`input languages <languages>`.
-The behavior of cvc5 can be changed with a number of :doc:`options <options>`, and :doc:`output tags <output-tags>` allow to extract structured information about the solving process.
+interface, but cvc5 also supports other input languages:
```
Comment (4tXJ7f) [inline docs/binary/binary.rst:6]:
```suggestion
interface.

The cvc5 binary supports the following input languages:
```

The "but ... also" sounds a bit weird in the original sentence.

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
Thanks for doing this. Just some minor comments.

**Review by 4tXJ7f — APPROVED** [review]
LGTM

**Pull request merged**